src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs

```diff
@@ -1413,9 +1627,13 @@ private static bool EnsureNumberOfValues(int segmentNumberOfValues, ref SingleRe
             // here we have a complex scenario, we need to add it in the middle of the current segment
             // to do that, we have to re-create it from scratch.
 
-            // the first thing to do here it to copy the segment out, because we may be writing it in multiple
-            // steps, and move the actual values as we do so
+            // the first thing to do here is trying to merge increment/decrement results to get a segment with unique entries
+            // then, we need to copy the segment out, because we may be writing it in multiple steps,
+            // and move the actual values as we do so
 
+            if (TryMergeIncrementResults(context, timeSeriesSegment))
```

Comment:
and if this didn't work?

Reply:
Do you refer to a false scenario? or to a failure in `TryMergeIncrementResults` (unhandled exception)? 
For the first - it probably means that this segment contains less than two live entries or nothing has changed after the merge, but it doesn't necessarily mean we can't continue with `SplitSegment` flow.